src/app/component-library/catalog/asset-detail-dialog/asset-detail-dialog-data.service.ts

```diff
@@ -24,11 +24,22 @@ export class AssetDetailDialogDataService {
     return {
       type: 'asset-details',
       asset,
-      showDeleteButton: allowDelete,
       propertyGridGroups,
     };
   }
 
+  assetDetailsEditable(
+    asset: UiAssetMapped,
+    opts: {onAssetEditClick: OnAssetEditClickFn},
```

Comment:
I have used it multiple times in this code base. it's about making a function accept a parameter object rather than individual parameters. this helps with optional parameters and especially things such as functions, where you really profit on the consuming side of having e.g. `someFunc({firstFunction: () => ..., secondFunction: () => ...})` rather than just having the lambdas in there without knowing what it does

Reply:
I just call it opts to try to keep things as short as possible in the code that uses it